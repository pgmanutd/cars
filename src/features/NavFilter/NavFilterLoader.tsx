import React from 'react';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';

import { useStyles } from './navFilterStyles';

export interface NavFilterLoaderProps {
  areColorsResponseLoading: boolean;
  areManufacturersResponseLoading: boolean;
  areNavFiltersResponseLoading: boolean;
}

const NavFilterLoader: React.FC<NavFilterLoaderProps> = ({
  areColorsResponseLoading,
  areManufacturersResponseLoading,
  areNavFiltersResponseLoading,
}) => {
  const classes = useStyles();

  return (
    <Box
      data-testid="NavFilterLoader"
      component="aside"
      className={classes.root}
    >
      {areColorsResponseLoading && (
        <Skeleton data-testid="NavFilterColorSkeleton" height={48} />
      )}
      {areManufacturersResponseLoading && (
        <Skeleton data-testid="NavFilterManufacturerSkeleton" height={48} />
      )}
      <Box className={classes.filterButtonContainer}>
        {areNavFiltersResponseLoading && (
          <Skeleton
            data-testid="NavFilterButtonSkeleton"
            height={70}
            width={92}
          />
        )}
      </Box>
    </Box>
  );
};

export default NavFilterLoader;
