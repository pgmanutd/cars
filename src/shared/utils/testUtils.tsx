import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
