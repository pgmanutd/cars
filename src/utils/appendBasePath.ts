const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
