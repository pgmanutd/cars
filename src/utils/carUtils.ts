import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
