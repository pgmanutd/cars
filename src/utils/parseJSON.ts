const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
