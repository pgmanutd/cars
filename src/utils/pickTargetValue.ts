import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
