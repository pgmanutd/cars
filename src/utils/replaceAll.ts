export type ReplaceAllMappedObject = { [key: string]: string };

const replaceAll = (
  value: string,
  mappedObject: ReplaceAllMappedObject = {},
) => {
  const keys = Object.keys(mappedObject);

  if (!keys.length) {
    return value;
  }

  const regex = new RegExp(keys.join('|'), 'gi');

  return value.replace(regex, (matched) => {
    return mappedObject[matched];
  });
};

export default replaceAll;
