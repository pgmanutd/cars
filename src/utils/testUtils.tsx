import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  ...render(
    <HelmetProvider>
      <Router history={history}>
        <Route
          path={path}
          render={routeProps => React.cloneElement(ui, { ...routeProps })}
        />
      </Router>
    </HelmetProvider>,
  ),
  history,
});
