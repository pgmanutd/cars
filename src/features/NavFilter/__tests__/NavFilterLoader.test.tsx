import React from 'react';

import { renderWithTranslate } from 'utils/testUtils';

import NavFilterLoader from '../NavFilterLoader';

describe('<NavFilterLoader />', () => {
  const setup = () => {
    const { renderResult } = renderWithTranslate(
      <NavFilterLoader
        areColorsResponseLoading
        areManufacturersResponseLoading
        areNavFiltersResponseLoading
      />,
    );

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
