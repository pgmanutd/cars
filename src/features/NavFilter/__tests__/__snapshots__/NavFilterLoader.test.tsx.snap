// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavFilterLoader /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <aside
    class="MuiBox-root MuiBox-root-6 makeStyles-root-1"
    data-testid="NavFilterLoader"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      data-testid="NavFilterColorSkeleton"
      style="height: 48px;"
    />
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      data-testid="NavFilterManufacturerSkeleton"
      style="height: 48px;"
    />
    <div
      class="MuiBox-root MuiBox-root-15 makeStyles-filterButtonContainer-4"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        data-testid="NavFilterButtonSkeleton"
        style="width: 92px; height: 70px;"
      />
    </div>
  </aside>
</DocumentFragment>
`;
